lib/lambda_ethereum_consensus/validator/validator.ex

```diff
@@ -32,220 +28,93 @@ defmodule LambdaEthereumConsensus.Validator do
-      slot: slot,
-      root: head_root
-    )
-
-    state
-  end
-
-  def handle_new_head(slot, head_root, state) do
-    log_debug(state.index, "recieved new head", slot: slot, root: head_root)
-
-    # TODO: this doesn't take into account reorgs
-    state
-    |> update_state(slot, head_root)
-    |> maybe_attest(slot)
-    |> maybe_build_payload(slot + 1)
-  end
-
-  @spec handle_tick({Types.slot(), atom()}, t()) :: t()
-  def handle_tick(_logical_time, %{index: nil} = state) do
-    log_error("-1", "setup validator", "index not present for handle tick")
-    state
-  end
-
-  def handle_tick({slot, :first_third}, state) do
-    log_debug(state.index, "started first third", slot: slot)
-    # Here we may:
-    # 1. propose our blocks
-    # 2. (TODO) start collecting attestations for aggregation
-    maybe_propose(state, slot)
-    |> update_state(slot, state.root)
-  end
-
-  def handle_tick({slot, :second_third}, state) do
-    log_debug(state.index, "started second third", slot: slot)
-    # Here we may:
-    # 1. send our attestation for an empty slot
-    # 2. start building a payload
-    state
-    |> maybe_attest(slot)
-    |> maybe_build_payload(slot + 1)
-  end
-
-  def handle_tick({slot, :last_third}, state) do
-    log_debug(state.index, "started last third", slot: slot)
-    # Here we may publish our attestation aggregate
-    maybe_publish_aggregate(state, slot)
-  end
-
-  ##########################
-  ### Private Functions
   ##########################
+  # Target State
 
-  @spec update_state(t(), Types.slot(), Types.root()) :: t()
-
-  defp update_state(%{slot: slot, root: root} = state, slot, root), do: state
-
-  # Epoch as part of the state now avoids recomputing the duties at every block
-  defp update_state(%{epoch: last_epoch} = state, slot, head_root) do
-    epoch = Misc.compute_epoch_at_slot(slot + 1)
-
-    if last_epoch == epoch do
-      %{state | slot: slot, root: head_root}
-    else
-      recompute_duties(state, last_epoch, epoch, slot, head_root)
-    end
+  @spec fetch_target_state_and_go_to_slot(Types.epoch(), Types.slot(), Types.root()) ::
+          Types.BeaconState.t()
+  def fetch_target_state_and_go_to_slot(epoch, slot, root) do
+    epoch |> fetch_target_state(root) |> go_to_slot(slot)
   end
 
-  @spec recompute_duties(t(), Types.epoch(), Types.epoch(), Types.slot(), Types.root()) :: t()
-  defp recompute_duties(%{root: last_root} = state, last_epoch, epoch, slot, head_root) do
-    start_slot = Misc.compute_start_slot_at_epoch(epoch)
-    target_root = if slot == start_slot, do: head_root, else: last_root
-
-    # Process the start of the new epoch
-    new_beacon = fetch_target_state(epoch, target_root) |> go_to_slot(start_slot)
-
-    new_duties =
-      Duties.shift_duties(state.duties, epoch, last_epoch)
-      |> Duties.maybe_update_duties(new_beacon, epoch, state.index, state.keystore.privkey)
-
-    move_subnets(state.duties, new_duties)
```

Comment:
This leaved subnets and joined the new ones. But in the new calculation of duties this isn't done and both publish/receive of attestations for aggregation is working. I'm going to create an issue for further check this.

Reply:
Added the issue: #1279